index.js

```diff
@@ -27,10 +27,15 @@ const HEADER = `// Type definitions for WebExtension Development in FireFox ${ar
 // TypeScript Version: 2.4
 // Generated using script at github.com/bomjacob/definitelytyped-firefox-webext-browser
 
-interface WebExtEventListener<T extends (...args: any[]) => any> {
-    addListener: (callback: T) => void;
-    removeListener: (listener: T) => void;
-    hasListener: (listener: T) => boolean;
+interface WebExtEventBase<TAddListener, TCallback> {
```

Comment:
maybe `TAddListener extends (...args: any[]) => void` could be added, i tried something similar for `TCallback` but i'd get an error even with just `extends Function`

Reply:
Yeah I'm unable to add any constraints to the TCallback too, I wonder if this is a bug somehow?